packages/wasm/crate/src/tx.rs

```diff
@@ -173,34 +161,53 @@ pub fn build_parallel(
     Ok(tx.encode_to_vec())
 }
 
-/// Get transaction view, transaction perspective
+/// Get transaction perspective, transaction view
 /// Arguments:
-///     full_viewing_key: `byte representation inner FullViewingKey`
-///     tx: `pbt::Transaction`
+///     full_viewing_key: `FullViewingKey` inner bytes
+///     tx: Binary-encoded `Transaction` message
 ///     idb_constants: IndexedDbConstants
-/// Returns: `TxInfoResponse`
-#[wasm_bindgen]
-pub async fn transaction_info(
+/// Returns: `{ txp: Uint8Array, txv: Uint8Array }` representing binary-encoded `TransactionPerspective` and `TransactionView`
+#[wasm_bindgen(typescript_custom_section)]
+const TRANSACTION_VIEW_PERSPECTIVE_TS: &'static str = r#"
+export function transaction_perspective_and_view(full_viewing_key: Uint8Array, tx: Uint8Array, idb_constants: any): Promise<{ txp: Uint8Array, txv: Uint8Array }>;
+"#;
+#[wasm_bindgen(skip_typescript)]
+pub async fn transaction_perspective_and_view(
     full_viewing_key: &[u8],
     tx: &[u8],
     idb_constants: JsValue,
-) -> WasmResult<JsValue> {
+) -> WasmResult<js_sys::Object> {
     utils::set_panic_hook();
 
     let transaction = Transaction::decode(tx)?;
     let constants = serde_wasm_bindgen::from_value(idb_constants)?;
     let fvk = FullViewingKey::decode(full_viewing_key)?;
-    let response = transaction_info_inner(fvk, transaction, constants).await?;
+    let (txp, txv) = transaction_info_inner(fvk, transaction, constants).await?;
 
-    let result = serde_wasm_bindgen::to_value(&response)?;
-    Ok(result)
+    let response = js_sys::Object::new();
+
+    js_sys::Reflect::set(
+        &response,
+        &"txp".into(),
+        &js_sys::Uint8Array::from(prost::Message::encode_to_vec(&txp).as_slice()),
+    )
+    .expect("Must set txp");
```

Comment:
We want to avoid .expect() as it can panic and causes an exception we can't handle